peps/pep-9999.rst

```diff
@@ -0,0 +1,548 @@
+Status: Draft
+Type: Process
+Created: 20-May-2024
+Python-Version: 3.26
+Post-History: <REQUIRED: dates, in dd-mmm-yyyy format, and corresponding links to PEP discussion threads>
+
+
+Abstract
+========
+
+This PEP proposes updating the versioning scheme for Python to include
+the calendar year.
+This aims to make the support lifecycle clear
+by making it easy to see when a version was first released,
+and easier to work out when it will reach end-of-life.
+
+Starting with what would have been Python 3.15,
+the version is 3.YY.micro where YY is the year of initial release:
+
+* Python 3.26 will be released in 2026 instead of Python 3.15.
+  EOL is five years after initial release,
+  therefore Python 3.26 will reach EOL in 2031.
+* Python 3.27 will be released in 2027, and so on.
+
+Motivation and rationale
+========================
+
+In 2019, we adopted an annual release cycle with :pep:`602`, which opened the
+door for calendar versioning:
+
+   Adopting an annual release calendar allows for natural switching to calendar
+   versioning, for example by calling Python 3.9 “Python 3.20” since it’s
+   released in October ‘20 and so on (“Python 3.23” would be the one released
+   in October ‘23).
+
+   While the ease of switching to calendar versioning can be treated as an
+   advantage of an annual release cycle, this PEP does not advocate for or
+   against a change in how Python is versioned. Should the annual release
+   cycle be adopted, the versioning question will be dealt with in a separate
+   PEP.
+
+This is that PEP.
+
+Current scheme
+--------------
+
+From the General Python FAQ:
+:external+python:ref:`faq-version-numbering-scheme`
+
+   Python versions are numbered "A.B.C" or "A.B":
+
+   * *A* is the major version number
+     -- it is only incremented for really major changes in the language.
+   * *B* is the minor version number
+     -- it is incremented for less earth-shattering changes.
+   * *C* is the micro version number
+     -- it is incremented for each bugfix release.
+
+Python predates SemVer
+----------------------
+
+`Semantic Versioning <https://semver.org/>`__ (SemVer)
+is a popular scheme which aims to communicate the intent of a release.
+
+   Given a version number MAJOR.MINOR.PATCH, increment the:
+
+   1. MAJOR version when you make incompatible API changes
+   2. MINOR version when you add functionality in a backward compatible manner
+   3. PATCH version when you make backward compatible bug fixes
+
+People often assume Python follows SemVer and complain about breaking changes
+in feature releases. But Python predates SemVer by at least 15 years:
+the SemVer spec was `introduced in 2009
+<https://github.com/semver/semver.org/commit/ca645805ca206e83c7153c64f9bda54afff06262>`__
+and the Python scheme was `added to source control in 1994
+<https://github.com/hugovk/cpython/commit/95f61a7ef067dbcabccc9b45ee885b0d55922c5f>`__
+for the 1.0 release.
```

Comment:
you can also add a reference how semver is just good intentions:
https://hynek.me/articles/semver-will-not-save-you/

Reply:
See https://github.com/pydantic/pydantic/issues/5821 for when I completely broke pydantic in a SemVer-compliant bugfix release of typing_extensions, which implemented a change that pydantic had asked for 🙃